src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php

```diff
@@ -104,4 +104,59 @@ protected function getDefinedElements()
             'last_order' => 'table tbody tr:last-child a:contains("Show")',
         ]);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCancelButtonVisibleForOrderWithNumber(string $number): bool
+    {
+        $orderData = $this->getSession()->getPage()->find('css', sprintf('tr:contains("%s")', $number));
+
+        if (null === $orderData) {
+            return false;
+        }
+
+        $actionButtonsText = $orderData->find('css', 'td:last-child')->getText();
+
+        return strpos($actionButtonsText, 'Cancel');
```

Comment:
There is lots of common logic in this and next method. Maybe we can extract private method that would get cancel button for order with number and throw exception if it does not exist? In this function we can catch this exception and return bool.